examples/gnn_cost_model/train_cost_model.py

```diff
@@ -0,0 +1,132 @@
+    dataset = CompilerGymDataset(file_dir, vocab=vocab, dataset_size=dataset_size)
+    train_data_loader = DataLoader(
+        dataset,
+        batch_size=batch_size,
+        shuffle=True,
+        num_workers=num_workers,
+        collate_fn=dataset.collate_fn,
+    )
+    return dataset, train_data_loader
+
+
+def dataset_looper(epoch_num, data_loader, model, device, optimizer=None, train=True):
+    times = collections.defaultdict(float)
+    losses = []
+    unscaled_mse = []
+    epoch_grad_clip = []
+    t1 = time.time()
+    for iter, data in enumerate(data_loader):
+        if data is None:
+            continue
+        graph, labels = data
+
+        times["get_data"] += time.time() - t1
+        t1 = time.time()
+
+        labels = torch.Tensor(labels).to(device)
+        graph = graph.to(device)
+        loss, unscaled, _ = model.get_loss(graph, labels)
+        losses.append(loss.cpu().data.numpy())
+        unscaled_mse.append(unscaled.cpu().data.numpy())
+        times["model_forward"] += time.time() - t1
+        t1 = time.time()
+
+        if train:
+            optimizer.zero_grad()
+            loss.backward()
+            grad_clip = torch.nn.utils.clip_grad_norm_(
+                model.parameters(), max_norm=400.0
+            )
+            epoch_grad_clip.append(grad_clip.cpu().data.numpy())
+            optimizer.step()
+
+            times["model_backward"] += time.time() - t1
+            t1 = time.time()
+    avg_loss, avg_unscaled, avg_grad_clip = (
+        np.mean(losses),
+        np.mean(unscaled_mse),
+        np.mean(epoch_grad_clip),
+    )
+    print(
+        f"Epoch num {epoch_num} training {train} took: {times}, loss: {avg_loss}, unscaled: {avg_unscaled}, grad_clip {avg_grad_clip}"
+    )
+
+    return avg_loss, avg_unscaled, avg_grad_clip
+
+
+def train(dataset, data_loader, model, num_epoch, device):
+    optimizer = torch.optim.Adam(model.parameters())
+    for epoch in range(num_epoch):
+        start_time = time.time()
+
+        dataset.set_distribution_type("train")
+        print("Running training with length of: ", len(dataset))
+        dataset_looper(epoch, data_loader, model, device, optimizer)
+        dataset.set_distribution_type("dev")
+        print("Running dev with length of: ", len(dataset))
+        dataset_looper(epoch, data_loader, model, device, train=False)
+
+        print("took: ", time.time() - start_time, "for an epoch")
+
+
+def main(argv):
+    """Main entry point."""
+    del argv  # unused
+    # root_pth = "/checkpoint/bcui/compiler_gym/replay_dataset/cbench/06-23/"
+    root_pth = (
+        "/checkpoint/bcui/compiler_gym/replay_dataset/frozen/2021-06-15-cbench-v1.db"
+    )
+    node_vocab_pth = "/checkpoint/bcui/compiler_gym/replay_dataset/cbench/06-23/2021-06-23-cbench-vocab.pkl"
+    batch_size = FLAGS.batch_size
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+    vocab = parsing_utils.load_vocab(node_vocab_pth)
```

Comment:
```suggestion
    vocab = pickle.load(open(node_vocab_pth, "rb"))
```

Then there's no need for `load_vocab()` func